sympy/printing/str.py

```
@@ -312,8 +312,48 @@ def _print_HadamardProduct(self, expr):
             for arg in expr.args])
 
     def _print_MatAdd(self, expr):
-        return ' + '.join([self.parenthesize(arg, precedence(expr))
-            for arg in expr.args])
+        # Mirror _print_Add behavior for matrices: render negative terms
+        # as subtraction rather than "+ -X" or explicit "(-1)*X".
+        # Preserve term order consistent with Add.
+        if self.order == 'none':
+            terms = list(expr.args)
+        else:
+            terms = self._as_ordered_terms(expr)
+
+        PREC = precedence(expr)
+        from sympy.matrices.expressions.matmul import MatMul
+        out = []
+        for i, term in enumerate(terms):
+            neg = False
+            pos_term = term
+            # Detect a leading negative coefficient in matrix products
+            if getattr(term, 'is_MatMul', False):
+                coeff, matrices = term.as_coeff_matrices()
+                try:
+                    is_neg = coeff.is_Number and coeff.is_negative
```

[major] This `try`/`except AttributeError` is a "just in case" fallback. `as_coeff_matrices()` always returns a SymPy scalar, and every SymPy scalar inherits `is_Number`/`is_negative`. If those attributes were missing we would want the code to fail loudly so we can fix the underlying bug, not silently treat the term as positive. Please drop the `try`/`except` and rely on the normal attribute access (and apply the same cleanup to the pretty/latex printers).